internal/contour/listener_test.go

```diff
@@ -550,12 +621,21 @@ func TestListenerVisit(t *testing.T) {
 			root := reh.Build()
 			got := visitListeners(root, &tc.ListenerVisitorConfig)
 			if !reflect.DeepEqual(tc.want, got) {
-				t.Fatalf("expected:\n%+v\ngot:\n%+v", tc.want, got)
+				weWant := prettyPrint(tc.want)
+				weGot := prettyPrint(got)
+				t.Fatal(cmp.Diff(weWant, weGot))
+				//t.Fatalf("expected:\n%+v\ngot:\n%+v", weWant, weGot)
 			}
 		})
 	}
 }
 
+// added to make the test output minimally readable.
+func prettyPrint(i interface{}) string {
+	s, _ := json.MarshalIndent(i, "", "\t")
+	return string(s)
+}
```

Comment:
I know the output from cmp.Diff isn't great, but the output from this hand rolled diff comparison doesn't look much better; 
```
--- FAIL: TestListenerVisit (0.00s)                                                                                                                                                                                                  [2/167]
    --- FAIL: TestListenerVisit/multiple_tls_ingress_with_secrets_should_be_sorted (0.00s)                                                                                                                                                  
        listener_test.go:626: {string}:                                                                                                                                                                                                     
                -: "{\n\t\"ingress_http\": {\n\t\t\"name\": \"ingress_http\",\n\t\t\"address\": {\n\t\t\t\"Address\": {\n\t\t\t\t\"SocketAddress\": {\n\t\t\t\t\t\"address\": \"0.0.0.0\",\n\t\t\t\t\t\"PortSpecifier\": {\n\t\t\t\t\t\t\"Po
rtValue\": 8080\n\t\t\t\t\t}\n\t\t\t\t}\n\t\t\t}\n\t\t},\n\t\t\"filter_chains\": [\n\t\t\t{\n\t\t\t\t\"filters\": [\n\t\t\t\t\t{\n\t\t\t\t\t\t\"name\": \"envoy.http_connection_manager\",\n\t\t\t\t\t\t\"ConfigType\": {\n\t\t\t\t\t\t\t\"T
ypedConfig\": {\n\t\t\t\t\t\t\t\t\"type_url\": \"type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager\",\n\t\t\t\t\t\t\t\t\"value\": \"EgxpbmdyZXNzX2h0dHAaIQoREg8IAiILCgkKB2NvbnRvdXISDGluZ3Jlc
3NfaHR0cCoMCgplbnZveS5nemlwKhAKDmVudm95LmdycGNfd2ViKg4KDGVudm95LnJvdXRlckICEAFaAgg8amUKFWVudm95LmZpbGVfYWNjZXNzX2xvZyJMCjt0eXBlLmdvb2dsZWFwaXMuY29tL2Vudm95LmNvbmZpZy5hY2Nlc3Nsb2cudjIuRmlsZUFjY2Vzc0xvZxINCgsvZGV2L3N0ZG91dHICCAHyAQIIAQ==\
"\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t}\n\t\t\t\t\t}\n\t\t\t\t]\n\t\t\t}\n\t\t],\n\t\t\"listener_filters\": null\n\t},\n\t\"ingress_https\": {\n\t\t\"name\": \"ingress_https\",\n\t\t\"address\": {\n\t\t\t\"Address\": {\n\t\t\t\t\"SocketAddres
s\": {\n\t\t\t\t\t\"address\": \"0.0.0.0\",\n\t\t\t\t\t\"PortSpecifier\": {\n\t\t\t\t\t\t\"PortValue\": 8443\n\t\t\t\t\t}\n\t\t\t\t}\n\t\t\t}\n\t\t},\n\t\t\"filter_chains\": [\n\t\t\t{\n\t\t\t\t\"filter_chain_match\": {\n\t\t\t\t\t\"ser
ver_names\": [\n\t\t\t\t\t\t\"sortedfirst.example.com\"\n\t\t\t\t\t]\n\t\t\t\t},\n\t\t\t\t\"tls_context\": {\n\t\t\t\t\t\"common_tls_context\": {\n\t\t\t\t\t\t\"tls_params\": {\n\t\t\t\t\t\t\t\"tls_minimum_protocol_version\": 2,\n\t\t\t
\t\t\t\t\"tls_maximum_protocol_version\": 4,\n\t\t\t\t\t\t\t\"cipher_suites\": [\n\t\t\t\t\t\t\t\t\"[ECDHE-ECDSA-AES128-GCM-SHA256|ECDHE-ECDSA-CHACHA20-POLY1305]\",\n\t\t\t\t\t\t\t\t\"[ECDHE-RSA-AES128-GCM-SHA256|ECDHE-RSA-CHACHA20-POLY
1305]\",\n\t\t\t\t\t\t\t\t\"ECDHE-ECDSA-AES128-SHA\",\n\t\t\t\t\t\t\t\t\"ECDHE-RSA-AES128-SHA\",\n\t\t\t\t\t\t\t\t\"ECDHE-ECDSA-AES256-GCM-SHA384\",\n\t\t\t\t\t\t\t\t\"ECDHE-RSA-AES256-GCM-SHA384\",\n\t\t\t\t\t\t\t\t\"ECDHE-ECDSA-AES256
-SHA\",\n\t\t\t\t\t\t\t\t\"ECDHE-RSA-AES256-SHA\"\n\t\t\t\t\t\t\t]\n\t\t\t\t\t\t},\n\t\t\t\t\t\t\"tls_certificate_sds_secret_configs\": [\n\t\t\t\t\t\t\t{\n\t\t\t\t\t\t\t\t\"name\": \"default/secret/735ad571c1\",\n\t\t\t\t\t\t\t\t\"sds_
config\": {\n\t\t\t\t\t\t\t\t\t\"ConfigSourceSpecifier\": {\n\t\t\t\t\t\t\t\t\t\t\"ApiConfigSource\": {\n\t\t\t\t\t\t\t\t\t\t\t\"api_type\": 2,\n\t\t\t\t\t\t\t\t\t\t\t\"grpc_services\": [\n\t\t\t\t\t\t\t\t\t\t\t\t{\n\t\t\t\t\t\t\t\t\t\t
\t\t\t\"TargetSpecifier\": {\n\t\t\t\t\t\t\t\t\t\t\t\t\t\t\"EnvoyGrpc\": {\n\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\"cluster_name\": \"contour\"\n\t\t\t\t\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t
\t\t\t\t\t]\n\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t],\n\t\t\t\t\t\t\"ValidationContextType\": null,\n\t\t\t\t\t\t\"alpn_protocols\": [\n\t\t\t\t\t\t\t\"h2\",\n\t\t\t\t\t\t\t\"http/1.
1\"\n\t\t\t\t\t\t]\n\t\t\t\t\t},\n\t\t\t\t\t\"SessionTicketKeysType\": null\n\t\t\t\t},\n\t\t\t\t\"filters\": [\n\t\t\t\t\t{\n\t\t\t\t\t\t\"name\": \"envoy.http_connection_manager\",\n\t\t\t\t\t\t\"ConfigType\": {\n\t\t\t\t\t\t\t\"Typed
Config\": {\n\t\t\t\t\t\t\t\t\"type_url\": \"type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager\",\n\t\t\t\t\t\t\t\t\"value\": \"Eg1pbmdyZXNzX2h0dHBzGiIKERIPCAIiCwoJCgdjb250b3VyEg1pbmdyZXNzX
2h0dHBzKgwKCmVudm95Lmd6aXAqEAoOZW52b3kuZ3JwY193ZWIqDgoMZW52b3kucm91dGVyQgIQAVoCCDxqZQoVZW52b3kuZmlsZV9hY2Nlc3NfbG9nIkwKO3R5cGUuZ29vZ2xlYXBpcy5jb20vZW52b3kuY29uZmlnLmFjY2Vzc2xvZy52Mi5GaWxlQWNjZXNzTG9nEg0KCy9kZXYvc3Rkb3V0cgIIAfIBAggB\"\n\
t\t\t\t\t\t\t}\n\t\t\t\t\t\t}\n\t\t\t\t\t}\n\t\t\t\t]\n\t\t\t},\n\t\t\t{\n\t\t\t\t\"filter_chain_match\": {\n\t\t\t\t\t\"server_names\": [\n\t\t\t\t\t\t\"sortedsecond.example.com\"\n\t\t\t\t\t]\n\t\t\t\t},\n\t\t\t\t\"tls_context\": {\n\
t\t\t\t\t\"common_tls_context\": {\n\t\t\t\t\t\t\"tls_params\": {\n\t\t\t\t\t\t\t\"tls_minimum_protocol_version\": 2,\n\t\t\t\t\t\t\t\"tls_maximum_protocol_version\": 4,\n\t\t\t\t\t\t\t\"cipher_suites\": [\n\t\t\t\t\t\t\t\t\"[ECDHE-ECDS
A-AES128-GCM-SHA256|ECDHE-ECDSA-CHACHA20-POLY1305]\",\n\t\t\t\t\t\t\t\t\"[ECDHE-RSA-AES128-GCM-SHA256|ECDHE-RSA-CHACHA20-POLY1305]\",\n\t\t\t\t\t\t\t\t\"ECDHE-ECDSA-AES128-SHA\",\n\t\t\t\t\t\t\t\t\"ECDHE-RSA-AES128-SHA\",\n\t\t\t\t\t\t\
t\t\"ECDHE-ECDSA-AES256-GCM-SHA384\",\n\t\t\t\t\t\t\t\t\"ECDHE-RSA-AES256-GCM-SHA384\",\n\t\t\t\t\t\t\t\t\"ECDHE-ECDSA-AES256-SHA\",\n\t\t\t\t\t\t\t\t\"ECDHE-RSA-AES256-SHA\"\n\t\t\t\t\t\t\t]\n\t\t\t\t\t\t},\n\t\t\t\t\t\t\"tls_certifica
te_sds_secret_configs\": [\n\t\t\t\t\t\t\t{\n\t\t\t\t\t\t\t\t\"name\": \"default/secret/735ad571c1\",\n\t\t\t\t\t\t\t\t\"sds_config\": {\n\t\t\t\t\t\t\t\t\t\"ConfigSourceSpecifier\": {\n\t\t\t\t\t\t\t\t\t\t\"ApiConfigSource\": {\n\t\t\t
\t\t\t\t\t\t\t\t\"api_type\": 2,\n\t\t\t\t\t\t\t\t\t\t\t\"grpc_services\": [\n\t\t\t\t\t\t\t\t\t\t\t\t{\n\t\t\t\t\t\t\t\t\t\t\t\t\t\"TargetSpecifier\": {\n\t\t\t\t\t\t\t\t\t\t\t\t\t\t\"EnvoyGrpc\": {\n\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\"clu
ster_name\": \"contour\"\n\t\t\t\t\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t\t\t]\n\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t],\n\t
\t\t\t\t\t\"ValidationContextType\": null,\n\t\t\t\t\t\t\"alpn_protocols\": [\n\t\t\t\t\t\t\t\"h2\",\n\t\t\t\t\t\t\t\"http/1.1\"\n\t\t\t\t\t\t]\n\t\t\t\t\t},\n\t\t\t\t\t\"SessionTicketKeysType\": null\n\t\t\t\t},\n\t\t\t\t\"filters\": [
\n\t\t\t\t\t{\n\t\t\t\t\t\t\"name\": \"envoy.http_connection_manager\",\n\t\t\t\t\t\t\"ConfigType\": {\n\t\t\t\t\t\t\t\"TypedConfig\": {\n\t\t\t\t\t\t\t\t\"type_url\": \"type.googleapis.com/envoy.config.filter.network.http_connection_ma
nager.v2.HttpConnectionManager\",\n\t\t\t\t\t\t\t\t\"value\": \"Eg1pbmdyZXNzX2h0dHBzGiIKERIPCAIiCwoJCgdjb250b3VyEg1pbmdyZXNzX2h0dHBzKgwKCmVudm95Lmd6aXAqEAoOZW52b3kuZ3JwY193ZWIqDgoMZW52b3kucm91dGVyQgIQAVoCCDxqZQoVZW52b3kuZmlsZV9hY2Nlc3Nf
bG9nIkwKO3R5cGUuZ29vZ2xlYXBpcy5jb20vZW52b3kuY29uZmlnLmFjY2Vzc2xvZy52Mi5GaWxlQWNjZXNzTG9nEg0KCy9kZXYvc3Rkb3V0cgIIAfIBAggB\"\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t}\n\t\t\t\t\t}\n\t\t\t\t]\n\t\t\t}\n\t\t],\n\t\t\"listener_filters\": [\n\t\t\t{\n\
t\t\t\t\"name\": \"envoy.listener.tls_inspector\",\n\t\t\t\t\"ConfigType\": null\n\t\t\t}\n\t\t]\n\t}\n}"                                                                                                                                   
                +: "{\n\t\"ingress_http\": {\n\t\t\"name\": \"ingress_http\",\n\t\t\"address\": {\n\t\t\t\"Address\": {\n\t\t\t\t\"SocketAddress\": {\n\t\t\t\t\t\"address\": \"0.0.0.0\",\n\t\t\t\t\t\"PortSpecifier\": {\n\t\t\t\t\t\t\"Po
rtValue\": 8080\n\t\t\t\t\t}\n\t\t\t\t}\n\t\t\t}\n\t\t},\n\t\t\"filter_chains\": [\n\t\t\t{\n\t\t\t\t\"filters\": [\n\t\t\t\t\t{\n\t\t\t\t\t\t\"name\": \"envoy.http_connection_manager\",\n\t\t\t\t\t\t\"ConfigType\": {\n\t\t\t\t\t\t\t\"T
ypedConfig\": {\n\t\t\t\t\t\t\t\t\"type_url\": \"type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager\",\n\t\t\t\t\t\t\t\t\"value\": \"EgxpbmdyZXNzX2h0dHAaIQoREg8IAiILCgkKB2NvbnRvdXISDGluZ3Jlc
3NfaHR0cCoMCgplbnZveS5nemlwKhAKDmVudm95LmdycGNfd2ViKg4KDGVudm95LnJvdXRlckICEAFaAgg8amUKFWVudm95LmZpbGVfYWNjZXNzX2xvZyJMCjt0eXBlLmdvb2dsZWFwaXMuY29tL2Vudm95LmNvbmZpZy5hY2Nlc3Nsb2cudjIuRmlsZUFjY2Vzc0xvZxINCgsvZGV2L3N0ZG91dHICCAHyAQIIAQ==\
"\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t}\n\t\t\t\t\t}\n\t\t\t\t]\n\t\t\t}\n\t\t],\n\t\t\"listener_filters\": null\n\t},\n\t\"ingress_https\": {\n\t\t\"name\": \"ingress_https\",\n\t\t\"address\": {\n\t\t\t\"Address\": {\n\t\t\t\t\"SocketAddres
s\": {\n\t\t\t\t\t\"address\": \"0.0.0.0\",\n\t\t\t\t\t\"PortSpecifier\": {\n\t\t\t\t\t\t\"PortValue\": 8443\n\t\t\t\t\t}\n\t\t\t\t}\n\t\t\t}\n\t\t},\n\t\t\"filter_chains\": [\n\t\t\t{\n\t\t\t\t\"filter_chain_match\": {\n\t\t\t\t\t\"ser
ver_names\": [\n\t\t\t\t\t\t\"sortedsecond.example.com\"\n\t\t\t\t\t]\n\t\t\t\t},\n\t\t\t\t\"tls_context\": {\n\t\t\t\t\t\"common_tls_context\": {\n\t\t\t\t\t\t\"tls_params\": {\n\t\t\t\t\t\t\t\"tls_minimum_protocol_version\": 2,\n\t\t\
t\t\t\t\t\"tls_maximum_protocol_version\": 4,\n\t\t\t\t\t\t\t\"cipher_suites\": [\n\t\t\t\t\t\t\t\t\"[ECDHE-ECDSA-AES128-GCM-SHA256|ECDHE-ECDSA-CHACHA20-POLY1305]\",\n\t\t\t\t\t\t\t\t\"[ECDHE-RSA-AES128-GCM-SHA256|ECDHE-RSA-CHACHA20-POL
Y1305]\",\n\t\t\t\t\t\t\t\t\"ECDHE-ECDSA-AES128-SHA\",\n\t\t\t\t\t\t\t\t\"ECDHE-RSA-AES128-SHA\",\n\t\t\t\t\t\t\t\t\"ECDHE-ECDSA-AES256-GCM-SHA384\",\n\t\t\t\t\t\t\t\t\"ECDHE-RSA-AES256-GCM-SHA384\",\n\t\t\t\t\t\t\t\t\"ECDHE-ECDSA-AES25
6-SHA\",\n\t\t\t\t\t\t\t\t\"ECDHE-RSA-AES256-SHA\"\n\t\t\t\t\t\t\t]\n\t\t\t\t\t\t},\n\t\t\t\t\t\t\"tls_certificate_sds_secret_configs\": [\n\t\t\t\t\t\t\t{\n\t\t\t\t\t\t\t\t\"name\": \"default/secret/735ad571c1\",\n\t\t\t\t\t\t\t\t\"sds
_config\": {\n\t\t\t\t\t\t\t\t\t\"ConfigSourceSpecifier\": {\n\t\t\t\t\t\t\t\t\t\t\"ApiConfigSource\": {\n\t\t\t\t\t\t\t\t\t\t\t\"api_type\": 2,\n\t\t\t\t\t\t\t\t\t\t\t\"grpc_services\": [\n\t\t\t\t\t\t\t\t\t\t\t\t{\n\t\t\t\t\t\t\t\t\t\
t\t\t\t\"TargetSpecifier\": {\n\t\t\t\t\t\t\t\t\t\t\t\t\t\t\"EnvoyGrpc\": {\n\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\"cluster_name\": \"contour\"\n\t\t\t\t\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\
t\t\t\t\t\t]\n\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t],\n\t\t\t\t\t\t\"ValidationContextType\": null,\n\t\t\t\t\t\t\"alpn_protocols\": [\n\t\t\t\t\t\t\t\"h2\",\n\t\t\t\t\t\t\t\"http/1
.1\"\n\t\t\t\t\t\t]\n\t\t\t\t\t},\n\t\t\t\t\t\"SessionTicketKeysType\": null\n\t\t\t\t},\n\t\t\t\t\"filters\": [\n\t\t\t\t\t{\n\t\t\t\t\t\t\"name\": \"envoy.http_connection_manager\",\n\t\t\t\t\t\t\"ConfigType\": {\n\t\t\t\t\t\t\t\"Type
dConfig\": {\n\t\t\t\t\t\t\t\t\"type_url\": \"type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager\",\n\t\t\t\t\t\t\t\t\"value\": \"Eg1pbmdyZXNzX2h0dHBzGiIKERIPCAIiCwoJCgdjb250b3VyEg1pbmdyZXNz
X2h0dHBzKgwKCmVudm95Lmd6aXAqEAoOZW52b3kuZ3JwY193ZWIqDgoMZW52b3kucm91dGVyQgIQAVoCCDxqZQoVZW52b3kuZmlsZV9hY2Nlc3NfbG9nIkwKO3R5cGUuZ29vZ2xlYXBpcy5jb20vZW52b3kuY29uZmlnLmFjY2Vzc2xvZy52Mi5GaWxlQWNjZXNzTG9nEg0KCy9kZXYvc3Rkb3V0cgIIAfIBAggB\"\n
\t\t\t\t\t\t\t}\n\t\t\t\t\t\t}\n\t\t\t\t\t}\n\t\t\t\t]\n\t\t\t},\n\t\t\t{\n\t\t\t\t\"filter_chain_match\": {\n\t\t\t\t\t\"server_names\": [\n\t\t\t\t\t\t\"sortedfirst.example.com\"\n\t\t\t\t\t]\n\t\t\t\t},\n\t\t\t\t\"tls_context\": {\n\
t\t\t\t\t\"common_tls_context\": {\n\t\t\t\t\t\t\"tls_params\": {\n\t\t\t\t\t\t\t\"tls_minimum_protocol_version\": 2,\n\t\t\t\t\t\t\t\"tls_maximum_protocol_version\": 4,\n\t\t\t\t\t\t\t\"cipher_suites\": [\n\t\t\t\t\t\t\t\t\"[ECDHE-ECDS
A-AES128-GCM-SHA256|ECDHE-ECDSA-CHACHA20-POLY1305]\",\n\t\t\t\t\t\t\t\t\"[ECDHE-RSA-AES128-GCM-SHA256|ECDHE-RSA-CHACHA20-POLY1305]\",\n\t\t\t\t\t\t\t\t\"ECDHE-ECDSA-AES128-SHA\",\n\t\t\t\t\t\t\t\t\"ECDHE-RSA-AES128-SHA\",\n\t\t\t\t\t\t\
t\t\"ECDHE-ECDSA-AES256-GCM-SHA384\",\n\t\t\t\t\t\t\t\t\"ECDHE-RSA-AES256-GCM-SHA384\",\n\t\t\t\t\t\t\t\t\"ECDHE-ECDSA-AES256-SHA\",\n\t\t\t\t\t\t\t\t\"ECDHE-RSA-AES256-SHA\"\n\t\t\t\t\t\t\t]\n\t\t\t\t\t\t},\n\t\t\t\t\t\t\"tls_certifica
te_sds_secret_configs\": [\n\t\t\t\t\t\t\t{\n\t\t\t\t\t\t\t\t\"name\": \"default/secret/735ad571c1\",\n\t\t\t\t\t\t\t\t\"sds_config\": {\n\t\t\t\t\t\t\t\t\t\"ConfigSourceSpecifier\": {\n\t\t\t\t\t\t\t\t\t\t\"ApiConfigSource\": {\n\t\t\t
\t\t\t\t\t\t\t\t\"api_type\": 2,\n\t\t\t\t\t\t\t\t\t\t\t\"grpc_services\": [\n\t\t\t\t\t\t\t\t\t\t\t\t{\n\t\t\t\t\t\t\t\t\t\t\t\t\t\"TargetSpecifier\": {\n\t\t\t\t\t\t\t\t\t\t\t\t\t\t\"EnvoyGrpc\": {\n\t\t\t\t\t\t\t\t\t\t\t\t\t\t\t\"clu
ster_name\": \"contour\"\n\t\t\t\t\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t\t\t]\n\t\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t\t}\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t],\n\t
\t\t\t\t\t\"ValidationContextType\": null,\n\t\t\t\t\t\t\"alpn_protocols\": [\n\t\t\t\t\t\t\t\"h2\",\n\t\t\t\t\t\t\t\"http/1.1\"\n\t\t\t\t\t\t]\n\t\t\t\t\t},\n\t\t\t\t\t\"SessionTicketKeysType\": null\n\t\t\t\t},\n\t\t\t\t\"filters\": [
\n\t\t\t\t\t{\n\t\t\t\t\t\t\"name\": \"envoy.http_connection_manager\",\n\t\t\t\t\t\t\"ConfigType\": {\n\t\t\t\t\t\t\t\"TypedConfig\": {\n\t\t\t\t\t\t\t\t\"type_url\": \"type.googleapis.com/envoy.config.filter.network.http_connection_ma
nager.v2.HttpConnectionManager\",\n\t\t\t\t\t\t\t\t\"value\": \"Eg1pbmdyZXNzX2h0dHBzGiIKERIPCAIiCwoJCgdjb250b3VyEg1pbmdyZXNzX2h0dHBzKgwKCmVudm95Lmd6aXAqEAoOZW52b3kuZ3JwY193ZWIqDgoMZW52b3kucm91dGVyQgIQAVoCCDxqZQoVZW52b3kuZmlsZV9hY2Nlc3Nf
bG9nIkwKO3R5cGUuZ29vZ2xlYXBpcy5jb20vZW52b3kuY29uZmlnLmFjY2Vzc2xvZy52Mi5GaWxlQWNjZXNzTG9nEg0KCy9kZXYvc3Rkb3V0cgIIAfIBAggB\"\n\t\t\t\t\t\t\t}\n\t\t\t\t\t\t}\n\t\t\t\t\t}\n\t\t\t\t]\n\t\t\t}\n\t\t],\n\t\t\"listener_filters\": [\n\t\t\t{\n\
t\t\t\t\"name\": \"envoy.listener.tls_inspector\",\n\t\t\t\t\"ConfigType\": null\n\t\t\t}\n\t\t]\n\t}\n}"  
```
vs go.cmp
```
--- FAIL: TestListenerVisit (0.00s)
    --- FAIL: TestListenerVisit/multiple_tls_ingress_with_secrets_should_be_sorted (0.00s)
        listener_test.go:623: {map[string]*v2.Listener}["ingress_https"]:
                -: s`name:"ingress_https" address:<socket_address:<address:"0.0.0.0" port_value:8443 > > filter_chains:<filter_chain_match:<server_names:"sortedfirst.example.com" > tls_context:<common_tls_context:<tls_params:<tls_minimu
m_protocol_version:TLSv1_1 tls_maximum_protocol_version:TLSv1_3 cipher_suites:"[ECDHE-ECDSA-AES128-GCM-SHA256|ECDHE-ECDSA-CHACHA20-POLY1305]" cipher_suites:"[ECDHE-RSA-AES128-GCM-SHA256|ECDHE-RSA-CHACHA20-POLY1305]" cipher_suites:"ECDHE
-ECDSA-AES128-SHA" cipher_suites:"ECDHE-RSA-AES128-SHA" cipher_suites:"ECDHE-ECDSA-AES256-GCM-SHA384" cipher_suites:"ECDHE-RSA-AES256-GCM-SHA384" cipher_suites:"ECDHE-ECDSA-AES256-SHA" cipher_suites:"ECDHE-RSA-AES256-SHA" > tls_certific
ate_sds_secret_configs:<name:"default/secret/735ad571c1" sds_config:<api_config_source:<api_type:GRPC grpc_services:<envoy_grpc:<cluster_name:"contour" > > > > > alpn_protocols:"h2" alpn_protocols:"http/1.1" > > filters:<name:"envoy.htt
p_connection_manager" typed_config:<type_url:"type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager" value:"\022\ringress_https\032\"\n\021\022\017\010\002\"\013\n\t\n\007contour\022\ringress_h
ttps*\014\n\nenvoy.gzip*\020\n\016envoy.grpc_web*\016\n\014envoy.routerB\002\020\001Z\002\010<je\n\025envoy.file_access_log\"L\n;type.googleapis.com/envoy.config.accesslog.v2.FileAccessLog\022\r\n\013/dev/stdoutr\002\010\001\362\001\002
\010\001" > > > filter_chains:<filter_chain_match:<server_names:"sortedsecond.example.com" > tls_context:<common_tls_context:<tls_params:<tls_minimum_protocol_version:TLSv1_1 tls_maximum_protocol_version:TLSv1_3 cipher_suites:"[ECDHE-EC
DSA-AES128-GCM-SHA256|ECDHE-ECDSA-CHACHA20-POLY1305]" cipher_suites:"[ECDHE-RSA-AES128-GCM-SHA256|ECDHE-RSA-CHACHA20-POLY1305]" cipher_suites:"ECDHE-ECDSA-AES128-SHA" cipher_suites:"ECDHE-RSA-AES128-SHA" cipher_suites:"ECDHE-ECDSA-AES25
6-GCM-SHA384" cipher_suites:"ECDHE-RSA-AES256-GCM-SHA384" cipher_suites:"ECDHE-ECDSA-AES256-SHA" cipher_suites:"ECDHE-RSA-AES256-SHA" > tls_certificate_sds_secret_configs:<name:"default/secret/735ad571c1" sds_config:<api_config_source:<
api_type:GRPC grpc_services:<envoy_grpc:<cluster_name:"contour" > > > > > alpn_protocols:"h2" alpn_protocols:"http/1.1" > > filters:<name:"envoy.http_connection_manager" typed_config:<type_url:"type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager" value:"\022\ringress_https\032\"\n\021\022\017\010\002\"\013\n\t\n\007contour\022\ringress_https*\014\n\nenvoy.gzip*\020\n\016envoy.grpc_web*\016\n\014envoy.routerB\002\020\001Z\002\010<je\n\025envoy.file_access_log\"L\n;type.googleapis.com/envoy.config.accesslog.v2.FileAccessLog\022\r\n\013/dev/stdoutr\002\010\001\362\001\002\010\001" > > > listener_filters:<name:"envoy.listener.tls_inspector" > `
                +: s`name:"ingress_https" address:<socket_address:<address:"0.0.0.0" port_value:8443 > > filter_chains:<filter_chain_match:<server_names:"sortedsecond.example.com" > tls_context:<common_tls_context:<tls_params:<tls_minimum_protocol_version:TLSv1_1 tls_maximum_protocol_version:TLSv1_3 cipher_suites:"[ECDHE-ECDSA-AES128-GCM-SHA256|ECDHE-ECDSA-CHACHA20-POLY1305]" cipher_suites:"[ECDHE-RSA-AES128-GCM-SHA256|ECDHE-RSA-CHACHA20-POLY1305]" cipher_suites:"ECDHE-ECDSA-AES128-SHA" cipher_suites:"ECDHE-RSA-AES128-SHA" cipher_suites:"ECDHE-ECDSA-AES256-GCM-SHA384" cipher_suites:"ECDHE-RSA-AES256-GCM-SHA384" cipher_suites:"ECDHE-ECDSA-AES256-SHA" cipher_suites:"ECDHE-RSA-AES256-SHA" > tls_certificate_sds_secret_configs:<name:"default/secret/735ad571c1" sds_config:<api_config_source:<api_type:GRPC grpc_services:<envoy_grpc:<cluster_name:"contour" > > > > > alpn_protocols:"h2" alpn_protocols:"http/1.1" > > filters:<name:"envoy.http_connection_manager" typed_config:<type_url:"type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager" value:"\022\ringress_https\032\"\n\021\022\017\010\002\"\013\n\t\n\007contour\022\ringress_https*\014\n\nenvoy.gzip*\020\n\016envoy.grpc_web*\016\n\014envoy.routerB\002\020\001Z\002\010<je\n\025envoy.file_access_log\"L\n;type.googleapis.com/envoy.config.accesslog.v2.FileAccessLog\022\r\n\013/dev/stdoutr\002\010\001\362\001\002\010\001" > > > filter_chains:<filter_chain_match:<server_names:"sortedfirst.example.com" > tls_context:<common_tls_context:<tls_params:<tls_minimum_protocol_version:TLSv1_1 tls_maximum_protocol_version:TLSv1_3 cipher_suites:"[ECDHE-ECDSA-AES128-GCM-SHA256|ECDHE-ECDSA-CHACHA20-POLY1305]" cipher_suites:"[ECDHE-RSA-AES128-GCM-SHA256|ECDHE-RSA-CHACHA20-POLY1305]" cipher_suites:"ECDHE-ECDSA-AES128-SHA" cipher_suites:"ECDHE-RSA-AES128-SHA" cipher_suites:"ECDHE-ECDSA-AES256-GCM-SHA384" cipher_suites:"ECDHE-RSA-AES256-GCM-SHA384" cipher_suites:"ECDHE-ECDSA-AES256-SHA" cipher_suites:"ECDHE-RSA-AES256-SHA" > tls_certificate_sds_secret_configs:<name:"default/secret/735ad571c1" sds_config:<api_config_source:<api_type:GRPC grpc_services:<envoy_grpc:<cluster_name:"contour" > > > > > alpn_protocols:"h2" alpn_protocols:"http/1.1" > > filters:<name:"envoy.http_connection_manager" typed_config:<type_url:"type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager" value:"\022\ringress_https\032\"\n\021\022\017\010\002\"\013\n\t\n\007contour\022\ringress_https*\014\n\nenvoy.gzip*\020\n\016envoy.grpc_web*\016\n\014envoy.routerB\002\020\001Z\002\010<je\n\025envoy.file_access_log\"L\n;type.googleapis.com/envoy.config.accesslog.v2.FileAccessLog\022\r\n\013/dev/stdoutr\002\010\001\362\001\002\010\001" > > > listener_filters:<name:"envoy.listener.tls_inspector" > `
```
We have #1123 to track the limitations of go-cmp. I don't think its worth hand rolling this special case given that the output isn't that awesome anyway.